diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/chapter3/EmptyBlockWarnTest.kt

```diff
@@ -133,15 +133,62 @@ class EmptyBlockWarnTest : LintTestBase(::EmptyBlock) {
     }
 
     @Test
+    @Tag(WarningNames.EMPTY_BLOCK_STRUCTURE_ERROR)
     fun `check empty lambda with config`() {
         lintMethod(
             """
-                    |fun foo() {
-                    |   val y = listOf<Int>().map {} 
-                    |}
-                """.trimMargin(),
+                |fun foo() {
+                |   val y = listOf<Int>().map {} 
+                |}
+            """.trimMargin(),
             LintError(2, 30, ruleId, "${EMPTY_BLOCK_STRUCTURE_ERROR.warnText()} different style for empty block", true),
             rulesConfigList = rulesConfigListEmptyBlockExist
         )
     }
+
+    @Test
+    fun `should not trigger on anonymous SAM classes #1`() {
+        lintMethod(
+            """
+                |fun foo() {
+                |   val proj = some.create(
+                |       Disposable {},
+                |       config
+                |   ).project
+                |}
+            """.trimMargin(),
+            rulesConfigList = rulesConfigListEmptyBlockExist
+        )
+    }
+
+    @Test
+    fun `should not trigger on anonymous SAM classes #2`() {
+        lintMethod(
+            """
+                |fun foo() {
+                |   val some = Disposable {}
```

Comment:
We also have suppressed warning in our code, where `Disposable` is used like this. Please find it (it's in `KotlinParser` probably) and unsuppress too.